amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/ConsumerApplication.java

```diff
@@ -0,0 +1,157 @@
+package software.amazon.kinesis.application;
+
+import static org.junit.Assume.assumeTrue;
+
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
+import software.amazon.kinesis.checkpoint.CheckpointConfig;
+import software.amazon.kinesis.common.ConfigsBuilder;
+import software.amazon.kinesis.common.InitialPositionInStreamExtended;
+import software.amazon.kinesis.config.KCLAppConfig;
+import software.amazon.kinesis.config.RetrievalMode;
+import software.amazon.kinesis.coordinator.CoordinatorConfig;
+import software.amazon.kinesis.coordinator.Scheduler;
+import software.amazon.kinesis.leases.LeaseManagementConfig;
+import software.amazon.kinesis.lifecycle.LifecycleConfig;
+import software.amazon.kinesis.metrics.MetricsConfig;
+import software.amazon.kinesis.processor.ProcessorConfig;
+import software.amazon.kinesis.retrieval.RetrievalConfig;
+import software.amazon.kinesis.utils.LeaseTableManager;
+import software.amazon.kinesis.utils.StreamExistenceManager;
+
+@Slf4j
+public class ConsumerApplication {
```

Comment:
Does this refactoring make `TestConsumer` redundant? Wanna make sure that we don't have the same implementation in two places. Happy for this can be a follow-up change if a lot of refactoring would be needed.

Reply:
As we discussed, addressing this with separate task